### ISSUE TITLE
Uploading files with non-ASCII names generate invalid download URLs

### ISSUE THREAD

**Issue by vjt** [ifad/colore#8]
When POSTing Colore a file with non-ASCII characters in the name, subsequent GETs to it generate not properly encoded URLs.

As reported by @algodave, `URI::InvalidURIError: URI must be ascii only "/document/icplex/993f43c9-25fd-4327-ace8-9943a476c377/Synth\u{e8}sredacteds\u{e9}.doc20160930-22915-4z769e"`

The problem lies in the JSON-style encoding used to generate the URL, whereas it should use percent-encoding.

**Comment by tagliala** [comment]
This issue could have been fixed by sinatra/sinatra#1412 for the file name in form parameters. As per the URL, we should provide fix at Rack level, because the URL has been already parsed in Sinatra. 

At the moment I'm not able to reproduce this issue

Rails was used to monkey patch URI::Parser before Ruby 2.6 to take in count encoding issues: https://github.com/rails/rails/blob/960ceb237ac886711b8d50cdca3ac13c2e147be8/activesupport/lib/active_support/core_ext/uri.rb#L5-L17

I suppose this is just for display purposes, but a proper configuration of the LANG variable on the server could help

```
# locale
locale: Cannot set LC_CTYPE to default locale: No such file or directory
locale: Cannot set LC_MESSAGES to default locale: No such file or directory
locale: Cannot set LC_ALL to default locale: No such file or directory
LANG=en_US.UTF-8
LANGUAGE=
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_PAPER="en_US.UTF-8"
LC_NAME="en_US.UTF-8"
LC_ADDRESS="en_US.UTF-8"
LC_TELEPHONE="en_US.UTF-8"
LC_MEASUREMENT="en_US.UTF-8"
LC_IDENTIFICATION="en_US.UTF-8"
LC_ALL=

# find . -type f -path './storage/*39/*'
./storage/docker/d6/39/.lock
./storage/docker/d6/39/v001/d??cker.docx
./storage/docker/d6/39/v001/_author.txt
./storage/docker/d6/39/metadata.json

# ls -al storage/docker/d6/39/v001/dōcker.docx 
-rw-r--r-- 1 root root 17448 Apr  8 13:20 'storage/docker/d6/39/v001/d'$'\305\215''cker.docx'
```

```
# locale
LANG=en_US.UTF-8
LANGUAGE=en_US.UTF-8
LC_CTYPE="C.UTF-8"
LC_NUMERIC="C.UTF-8"
LC_TIME="C.UTF-8"
LC_COLLATE="C.UTF-8"
LC_MONETARY="C.UTF-8"
LC_MESSAGES="C.UTF-8"
LC_PAPER="C.UTF-8"
LC_NAME="C.UTF-8"
LC_ADDRESS="C.UTF-8"
LC_TELEPHONE="C.UTF-8"
LC_MEASUREMENT="C.UTF-8"
LC_IDENTIFICATION="C.UTF-8"
LC_ALL=C.UTF-8

# find . -type f -path './storage/*39/*'
./storage/docker/d6/39/.lock
./storage/docker/d6/39/v001/dōcker.docx
./storage/docker/d6/39/v001/_author.txt
./storage/docker/d6/39/metadata.json

# ls -al storage/docker/d6/39/v001/dōcker.docx 
-rw-r--r-- 1 root root 17448 Apr  8 13:20 storage/docker/d6/39/v001/dōcker.docx
```

Ref:
- https://github.com/whitequark/rack-utf8_sanitizer

**Comment by vjt** [comment]
Closing then. Thanks!
